matlab_simulation/code v1.0/ME597-8-Voronoi/Voronoi.m

```diff
@@ -1,11 +1,96 @@
 %% Voronoi Diagram
+clear; clc; 
+
+%% Create AVI object
+vidObj = VideoWriter('voronoi.avi');
+vidObj.Quality = 100;
+vidObj.FrameRate = 5;
+open(vidObj);
+
 
 %% Generate a set of point obstacles
-n = 100; % Points
-s = 100; % Space
+n = 300; % Points
+s = 1000; % Space
+
+%rng('default')
 
 % Random nodes
 points = s*rand(n,2);
 figure(1);clf;
-voronoi(points(:,1),points(:,2));
+plot(points);
+dt = delaunayTriangulation(points(:,1),points(:,2));
+H = voronoi(dt);
+[V,C] = voronoiDiagram(dt);
+writeVideo(vidObj, getframe(gcf));
+
+% Generate start and end locations
+startLocation = s*rand(1,2);
+endLocation = s*rand(1,2);
+%Closes Voronoi vertex to the end and start locations
+closestToStart = dsearchn(V, startLocation);
+closestToEnd = dsearchn(V, endLocation);
+
+writeVideo(vidObj, getframe(gcf));
+hold on;
+
+plot(V(closestToStart,1),V(closestToStart,2),'ro','MarkerSize',6,'LineWidth',2);
+writeVideo(vidObj, getframe(gcf));
+
+plot(V(closestToEnd,1),V(closestToEnd,2),'ro','MarkerSize',6,'LineWidth',2);
+writeVideo(vidObj, getframe(gcf));
+
+%% Initialize Robot
+l = 10;
+r = 5;
+robotWidth = 2*l;
+wheelVelocity = [];
+
+%% Create graph with edges of the Voronoi Diagram
+G = graph;
+G = addnode(G,length(V));
+
+for i=1:length(C)
+    for k=1:(length(C{i})-1)
+        idxOut = findedge(G,C{i}(k),C{i}(k+1));
+        if(idxOut == 0)
+            G = addedge(G, C{i}(k), C{i}(k+1), distanceTwoPoints(V(C{i}(k),:),V(C{i}(k+1),:)));    
+        end; 
+    end;
+end;
+
```

Comment:
In order to increase code easy to read, maybe it could be excluded in function i.e create_graph(G,V,C) ?